reference-library/TransientStateLibrary.md

```diff
@@ -0,0 +1,122 @@
+
+Common operations that involve transient state include:
+
+- Checking reserves (`getReserves`)
+- Verifying currency deltas (`currencyDelta`)
+- Syncing currency states (`sync`)
+- Settling currency balances (`settle`)
+
+### Storage Slots
+
+The library defines constant storage slots for different pieces of transient state:
+
+``` solidity
+bytes32 public constant RESERVES_OF_SLOT = 0x1e0745a7db1623981f0b2a5d4232364c00787266eb75ad546f190e6cebe9bd95;
+bytes32 public constant NONZERO_DELTA_COUNT_SLOT = 0x7d4b3164c6e45b97e7d87b7125a44c5828d005af88f9d751cfd78729c5d99a0b;
+bytes32 public constant IS_UNLOCKED_SLOT = 0xc090fc4683624cfc3884e9d8de5eca132f2d0ec062aff75d43c0465d5ceeab23;
+```
+
+These slots are computed using `keccak256` hashes of specific strings, ensuring unique storage locations for each piece of data.
+
+## Functions
+
+### getReserves
+
+``` solidity
+function getReserves(IPoolManager manager, Currency currency) internal view returns (uint256)
+```
+
+Retrieves the reserves of a specific currency from the PoolManager's transient storage.
+
+| Param Name   | Type         | Description                          |
+|--------------|--------------|--------------------------------------|
+| manager | IPoolManager | The PoolManager contract instance
+| currency | Currency | The currency for which to fetch reserves             |
+
+**Returns:**
+- `uint256`: The amount of reserves for the specified currency
+
+**Notes:**
+- Returns `0` if the reserves are not synced
+- Returns `type(uint256).max` if the reserves are synced but the value is `0`
+
+### getNonzeroDeltaCount
+
+```solidity
+function getNonzeroDeltaCount(IPoolManager manager) internal view returns (uint256)
+```
+
+Retrieves the count of nonzero deltas that must be zeroed out before the contract can be locked.
+
+| Param Name   | Type         | Description                          |
+|--------------|--------------|--------------------------------------|
+| manager | IPoolManager | The PoolManager contract instance
+
+**Returns:**
+
+- `uint256`: The number of nonzero deltas
+
+### currencyDelta
+
+```solidity
+function currencyDelta(IPoolManager manager, address caller_, Currency currency) internal view returns (int256)
+```
+
+Fetches the current delta for a specific caller and currency from the PoolManager's transient storage.
+
+| Param Name   | Type         | Description                          |
+|--------------|--------------|--------------------------------------|
+| manager | IPoolManager | The PoolManager contract instance
+| caller_ | address | The address of the caller
+| currency | Currency | The currency for which to lookup the delta             |
+
+**Returns:**
+
+- `int256`: The delta value for the specified caller and currency
```

Comment:
can we document that a negative delta is an amount that must be paid/settled, and a postive delta is an amount that must be taken